spec/lib/internal_metric/global_spec.rb

```diff
@@ -112,12 +112,14 @@ module DiscoursePrometheus::InternalMetric
 
     describe 'when a replica has been configured' do
       before do
-        config = ActiveRecord::Base.connection_config
+        config = ActiveRecord::Base.connection_db_config.configuration_hash.dup
 
         config.merge!(
           replica_host: 'localhost',
           replica_port: 1111
         )
+        ActiveRecord::Base.connection.disconnect!
+        ActiveRecord::Base.establish_connection(config)
```

Comment:
Not a big fan of this, but Rails 6.1 freezes the connection config hash. I could cheat using `instance_variable_set` and change the underlying hash inside `db_config`, but that doesn't seem like a great solution either.